dmcontent/govuk_frontend.py

```diff
@@ -107,11 +107,30 @@ def from_question(
 def govuk_input(
     question: Question, data: Optional[dict] = None, errors: Optional[dict] = None, **kwargs
 ) -> dict:
-    """Create govukInput macro parameters from a text question"""
+    """Create govukInput macro parameters from a text or number question"""
 
     params = _params(question, data, errors)
     params["classes"] = "app-text-input--height-compatible"
 
+    if question.type == "number":
+        params["classes"] += " govuk-input--width-5"
+        params["spellcheck"] = False
+        if question.get("limits") and question.limits.get("integer_only") is True:
+            params["inputmode"] = "numeric"
+            params["pattern"] = "[0-9]*"
+        else:
+            params["inputmode"] = "text"
```

Comment:
Do we ever ask for decimal numbers?

If so, we'd have to not include `inputmode` and `pattern` as per https://design-system.service.gov.uk/components/text-input/#asking-for-decimal-numbers

Reply:
Yeah, decimal numbers is the `else` branch. In the schema for our question YAML files it is assumed to be decimal unless `integer_only` is true-ish https://github.com/alphagov/digitalmarketplace-frameworks/blob/master/schema_generator/validation.py#L374